core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -197,7 +203,7 @@ func (r *EvmRegistry) allowedToUseMercury(opts *bind.CallOpts, upkeepId *big.Int
 	if err != nil {
 		return false, fmt.Errorf("failed to unmarshal privilege config for upkeep ID %s: %v", upkeepId, err)
 	}
-	r.mercury.allowListCache.Set(upkeepId.String(), a.MercuryEnabled, cache.DefaultExpiration)
+	r.mercury.allowListCache.Set(au.Admin.Hex(), a.MercuryEnabled, cache.DefaultExpiration)
```

Comment:
nitpick here but I think we can do better than "a" as a variable name! :stuck_out_tongue: